src/ORM/Search/SearchContextForm.php

```diff
@@ -0,0 +1,210 @@
+<?php
+
+namespace SilverStripe\ORM\Search;
+
+use SilverStripe\Control\RequestHandler;
+use SilverStripe\Core\ClassInfo;
+use SilverStripe\Forms\CompositeField;
+use SilverStripe\Forms\FieldList;
+use SilverStripe\Forms\Form;
+use SilverStripe\Forms\GridField\GridField;
+use SilverStripe\Forms\GridField\GridField_FormAction;
+use SilverStripe\Forms\GridField\GridFieldFilterHeader;
+use SilverStripe\ORM\FieldType\DBHTMLText;
+use SilverStripe\View\SSViewer;
+
+/**
+ * A form for using a search context in the CMS.
+ *
+ * Note that the form submission goes through to the index action of the controller by default,
+ * and is intended to return the same markup you would ordinarily get for that controller but with
+ * the results filtered appropriately.
+ */
+class SearchContextForm extends Form
+{
+    private SearchContext $searchContext;
+
+    private ?string $searchField = null;
+
+    private static $casting = [
+        'SchemaData' => 'Text',
+        'getSchemaData' => 'Text',
+    ];
+
+    public function __construct(RequestHandler $controller, SearchContext $searchContext, $name = 'SearchForm')
+    {
+        $this->searchContext = $searchContext;
+        $searchFields = clone $searchContext->getSearchFields();
```

Comment:
Why do we need to clone the search fields here?

Reply:
OK, just add that in an inline comment above so future generation know why it's cloned

Reply:
Done